torch/_subclasses/meta_utils.py

```diff
@@ -247,6 +249,58 @@ def sym_sizes_strides_storage_offset(t):
                         with torch.enable_grad():
                             r = r.clone()
                             r._coalesced_(t.is_coalesced())
+                elif t.is_nested:
+                    assert (
+                        shape_env is not None
+                    ), "Dynamic shapes must be enabled for nested tensors"
+                    is_leaf = safe_is_leaf(t)
+
+                    from torch._dynamo.source import AttrSource
+
+                    sizes = t._nested_tensor_size()
+                    sizes = self.meta_tensor(
+                        sizes,
+                        shape_env=shape_env,
+                        callback=callback,
+                        source=AttrSource(source, "_nested_tensor_size"),
+                    )
+
+                    strides = t._nested_tensor_strides()
+                    strides = self.meta_tensor(
+                        strides,
+                        shape_env=shape_env,
+                        callback=callback,
+                        source=AttrSource(source, "_nested_tensor_strides"),
+                    )
+
+                    offsets = t._nested_tensor_storage_offsets()
+                    offsets = self.meta_tensor(
+                        offsets,
+                        shape_env=shape_env,
+                        callback=callback,
+                        source=AttrSource(source, "_nested_tensor_storage_offsets"),
+                    )
+
+                    from torch.fx.experimental.symbolic_shapes import constrain_range
+
+                    buffer_size = shape_env.create_unbacked_symint()
+                    constrain_range(buffer_size, min=2, max=sys.maxsize - 1)
```

Comment:
O,1 shape specializing I presume but what if someone does operate on singular element nts ?

Reply:
I have the same thought but I'm not sure. Is this a hard range? @ezyang

Reply:
The idea is that nested tensor is expected to behave uniformly across all numbers, so if you get a trace for buffer size 2, it will also work for buffer size 1. This is an unchecked assumption and you'll have to verify the model does the right thing independently.